pkg/controller/appdefinition/deploy.go

```diff
@@ -597,12 +597,34 @@ func containerAnnotation(container v1.Container) string {
 	return string(json)
 }
 
+func resolvedOfferingsAnnotation(appInstance *v1.AppInstance, container v1.Container) (string, error) {
+	if resolved, exists := appInstance.Status.ResolvedOfferings.Containers[container.Name]; exists {
+		data, err := convert.EncodeToMap(resolved)
+		if err != nil {
+			return "", err
+		}
+
+		j, err := json.Marshal(data)
+		if err != nil {
+			return "", err
+		}
+
+		return string(j), nil
+	}
+	return "", nil
+}
+
 func podAnnotations(appInstance *v1.AppInstance, container v1.Container) map[string]string {
 	annotations := map[string]string{
 		labels.AcornContainerSpec: containerAnnotation(container),
 	}
 	addPrometheusAnnotations(annotations, container)
 
+	offerings, err := resolvedOfferingsAnnotation(appInstance, container)
```

Comment:
Looks like this error is still not handled...

Reply:
Below this line there is a check for `if err == nil`, and then we attach the annotation. I don't actually want to return this error if it happens, since we can safely just ignore it and keep going.